docs/how-to/integrate-external-libraries.md

```diff
@@ -0,0 +1,252 @@
+> [!NOTE]
+> Some libraries may not build properly with this method. OpenCV, for example, [does not support it](https://github.com/opencv/opencv/issues/26955#issuecomment-2690702771) due to the complexity of its CMake set up (as of April 2025). In such cases, you will need to use `find_package()` or `ExternalProject_Add` as described in the sections below.
+
+
+##### Step 1: Use add_subdirectory (or equivalent FetchContent) to include the library in your project
+
+The following excerpt demonstrates how to integrate the [ETL libraries](https://github.com/ETLCPP/etl) using `FetchContent`:
+
+```cmake
+# This code can be located in the root project.cmake file if the library is used by multiple modules
+# or directly in the module's CMakeLists.txt if it is only used by that module.
+include(FetchContent)
+FetchContent_Declare(
+    etl
+    GIT_REPOSITORY https://github.com/ETLCPP/etl
+    GIT_TAG        20.40.0
+)
+FetchContent_MakeAvailable(etl)
+```
+
+In the `fprime-examples` repository, this is done in [ExternalLibs/CMakeLists.txt](https://github.com/nasa/fprime-examples/tree/devel/FlightExamples/ExternalLibs/CMakeLists.txt) file. This allows any module within the ExternalLibs module to use `etl` library.
+If the library is only needed by a single component, it is also acceptable to place the `FetchContent` code directly in that component's `CMakeLists.txt` file. If the library is needed at the project level, it is best to place the `FetchContent` code in the root `project.cmake` file.
+
+##### Step 2: Set the library as a dependency of the module
+
+Any component or module that depends on ETL can now link against it by adding the following entry to its `MOD_DEPS`:
+
+```cmake
+# In MyComponent/CMakeLists.txt 
+set(SOURCE_FILES
+  "${CMAKE_CURRENT_LIST_DIR}/MyComponent.fpp"
+  "${CMAKE_CURRENT_LIST_DIR}/MyComponent.cpp"
+)
+set(MOD_DEPS
+  etl::etl   # This is the target name for the ETL library
+)
+register_fprime_module()
+```
+
+---
+
+## Approach 3: Install once and integrate with find_package()
+
+|     |     |
+| --- | --- |
+| **Benefits** | Easier than approach 4 |
+| **Drawbacks** | Additional step managed outside of CMake |
+| **Considerations** | Need to ensure developers will install the dependency |
+
+Some libraries may not be built with CMake but still integrate with CMake through [installed config files](https://cmake.org/cmake/help/v3.31/guide/using-dependencies/index.html). In this case, you may choose to install the library manually on your machine and use [`find_package()`](https://cmake.org/cmake/help/v3.31/command/find_package.html) in CMake to locate it. This is a common approach for libraries that have complex build requirements.
+
+##### Step 1: Install the library on your system
+
+There can be many ways to install the library, such as using a package manager (e.g., `apt`, `brew`, `yum`, `conda`, etc.), or building from source. Refer to each the library installation instructions to decide which fits your project best.
```

Comment:
Add a note about ensuring proper cross-compilation, or using a sysroot to provide the library compiled for your target system.